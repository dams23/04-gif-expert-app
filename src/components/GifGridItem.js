import React from 'react'
import PropTypes from 'prop-types'


const GifGridItem = ({ id, title, url }) => {
    return (
        <div className='card animate__animated animate__fadeInDownBig'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
GifGridItem.propTypes = {
    title : PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};
export default GifGridItem;