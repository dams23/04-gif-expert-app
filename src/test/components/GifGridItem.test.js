import React from 'react'
import  GifGridItem  from '../../components/GifGridItem'
import { shallow } from 'enzyme'

describe('Pruebas en el Item', () => {
    const title = 'Un titulo'
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(< GifGridItem 
        title={title} 
        url={url}
        />)
    test('debe mostrar el GifGridItem', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de tener un parrfo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
    test('debe de tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className') 
        expect(className.includes('animate__fadeInDownBig')).toBe(true);
    });
    
    
})
