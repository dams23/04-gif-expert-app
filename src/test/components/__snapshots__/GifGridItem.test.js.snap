// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Item debe mostrar el GifGridItem 1`] = `
<div
  className="card animate__animated animate__fadeInDownBig"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
